src/include/storage/write_ahead_log/log_manager.h

```diff
@@ -28,15 +33,51 @@ class LogManager {
    *                    buffers from
    */
   LogManager(const char *log_file_path, RecordBufferSegmentPool *const buffer_pool)
-      : out_(log_file_path), buffer_pool_(buffer_pool) {}
+      : buffer_pool_(buffer_pool),
+        log_file_path_(log_file_path),
+        buffer_to_write_(nullptr),
+        log_writer_(nullptr),
+        run_log_writer_thread_(false),
+        do_persist_(true) {}
+
+  /**
+   * Start logging
+   */
+  void Start() {
+    // Initialize buffers for logging
+    for (int i = 0; i < MAX_BUF; i++) {
+      buffers_.emplace_back(BufferedLogWriter(log_file_path_));
+    }
+    for (int i = 0; i < MAX_BUF; i++) {
+      empty_buffer_queue_.Enqueue(&buffers_[i]);
+    }
+    run_log_writer_thread_ = true;
+    log_writer_ = new LogWriter(this);
+  }
 
   /**
    * Must be called when no other threads are doing work
    */
   void Shutdown() {
+    // Process the remaining redo buffers
```

Comment:
I don't think the LogManager should own the log writer thread. As we have discussed,  we want a centralized registry for all these threads for self-driving purposes.

Reply:
As of now, what do you suggest would be the right way to do this?